### DISCUSSION TITLE
Runtime identifier to use when publishing iOS package

### DISCUSSION THREAD

**Discussion by Juriyx** [unoplatform/uno#16866] [Q&A]
Hello. I used this command to build a single Uno project for all platforms:
`msbuild /r /p:Configuration=$(BuildConfiguration)`
In result, for iOS I get a folder for this framework `iossimulator-x64`, but I want to get `ios-arm64`. How to do that? So that I build once with single build command.

![image](https://github.com/unoplatform/uno/assets/10002290/bfd8e17e-cf67-4282-9bc5-0b2a9cf81e15)

**Comment by agneszitte** [comment]
cc @jeromelaban, @spouliot

**Comment by spouliot** [comment]
I don't think you can [1] publish for all platforms inside a single project. Without any additional logic any argument you give to `msbuild` is used for all platforms. 

The trouble is that some defaults (made way earlier than single projects), like the iOS target for simulators, are not _ideal_ to publish an app. Changing them _globally_ (for all platforms) might work or might cause issues - that all depends on each argument.

To avoid this problem I suggest you to create a script that individually builds each platform you support. This way you can provide explicit arguments for each platform, without any fear they could confuse/contaminate other platforms.

[1] unless you add some msbuild logic inside your project

**Reply by jeromelaban** [reply]
@Juriyx note that publishing apps for iOS is better done on a mac directly, [using these steps](https://platform.uno/docs/articles/uno-publishing-ios.html#packaging-your-app-using-the-cli).
